Prepare release notes for v6.1.1 (#3568)
dotnet/SqlClient: release-notes/6.1/6.1.1.md

# Release Notes

## Stable Release 6.1.0 - 2025-07-25

This update brings the following changes since the
[6.1.0](6.1.0.md) release:

### Fixed

> - Reverted several large changes made to improve partial packet detection, fixup, and replay functionality to address regressions introduced in 6.1.0.

Reviewer: ```suggestion
- Reverted the packet multiplexing design changes that caused several regressions in 6.1.0.
```

  ([#3556](https://github.com/dotnet/SqlClient/pull/3556))
- Corrected reference assemblies for vector support and re-enabled vector and JSON tests to prevent API surface mismatches.
  ([#3562](https://github.com/dotnet/SqlClient/pull/3562))

### Changed

> #### Revert partial-packet logic and replay fixes

Reviewer: Don't need any sections actually.

 ([#3562](https://github.com/dotnet/SqlClient/pull/3562))

*Who Benefits:*

- Developers relying on updated vector support or testing frameworks benefit from consistent builds.

*Impact:*

- No end-user behavior change; improves CI and test reliability.

#### Upgraded `Azure.Identity` and other dependencies to newer versions.

*What Changed:*

- Upgraded `Azure.Identity` and other dependencies to newer versions.
 ([#3538](https://github.com/dotnet/SqlClient/pull/3538))
 ([#3552](https://github.com/dotnet/SqlClient/pull/3552))

*Who Benefits:*

-  Users gain compatibility with recent identity and auth packages, including security and performance patches in dependencies.

*Impact:*

- Slight potential for behavior differences tied to third-party dependency updates.

#### Other Changes

- Fix API documentation.
  ([#3533](https://github.com/dotnet/SqlClient/pull/3533))
- Add CodeQL suppressions for DefaultAzureCredential.
  ([#3542](https://github.com/dotnet/SqlClient/pull/3542),
   [#3550](https://github.com/dotnet/SqlClient/pull/3550))
- Fix project dependencies.
  ([#3561](https://github.com/dotnet/SqlClient/pull/3561))

> 

Reviewer: The API signature fix got missed out.
